src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -4267,6 +4268,32 @@ what() const noexcept
   return handle->m_message.c_str();
 }
 
+static std::string
+amend_aie_error_message(const ert_packet* epkt, const std::string& msg)
+{
+  std::ostringstream oss;
+  static const std::map<uint32_t, const char*> fatal_error_string {
+    {0, "N/A"}
+  };
+
+  if (epkt->state != ERT_CMD_STATE_TIMEOUT)
+    return msg;
+
+  oss << msg << "\n";
```

Comment:
No need to create the stringstream unless we need it.

```suggestion
  static const std::map<uint32_t, const char*> fatal_error_string {
    {0, "N/A"}
  };

  if (epkt->state != ERT_CMD_STATE_TIMEOUT)
    return msg;
    
  std::ostringstream oss;
  oss << msg << "\n";
```